internal/impl/gcp/input_spanner_change_stream.go

```diff
@@ -0,0 +1,277 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcp
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"slices"
+	"time"
+
+	"cloud.google.com/go/spanner"
+	"github.com/Jeffail/shutdown"
+	"github.com/anicoll/screamer/pkg/model"
+	"github.com/anicoll/screamer/pkg/partitionstorage"
+	"github.com/anicoll/screamer/pkg/screamer"
+	"github.com/redpanda-data/benthos/v4/public/service"
+)
+
+type streamReader interface {
+	Stream(ctx context.Context, channel chan<- *model.DataChangeRecord) error
+	Close() error
+}
+
+func newSpannerChangeStreamInputConfig() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		Beta().
+		Version("3.43.0").
+		Categories("Services", "GCP").
+		Summary("Creates an input that consumes from a spanner change stream.").
+		Field(service.NewStringField("stream_dsn").Description("Required field to use to connect to spanner for the change stream.").Example("projects/<project_id>/instances/<instance_id>/databases/<database_id>")).
```

Comment:
nit: we are now creating the fields as constants so it's harder to mess them up.